component/core/src/main/java/org/exoplatform/social/core/listeners/GroupSynchronizationSocialProfileListener.java

```diff
@@ -0,0 +1,164 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.exoplatform.social.core.listeners;
+
+import org.apache.commons.lang3.StringUtils;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.*;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.profile.ProfileLifeCycleEvent;
+import org.exoplatform.social.core.profile.ProfileListenerPlugin;
+import org.exoplatform.social.core.profile.settings.ProfilePropertySettingsService;
+import org.exoplatform.social.core.profileproperty.model.ProfilePropertySetting;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class GroupSynchronizationSocialProfileListener extends ProfileListenerPlugin {
+
+  private static final Log                     LOG                =
+                                                   ExoLogger.getLogger(GroupSynchronizationSocialProfileListener.class);
+
+  private static final String                  PROFILE_GROUP_NAME = "profile";
+
+  private static final String                  MEMBER             = "member";
+
+  private final ProfilePropertySettingsService profilePropertySettingsService;
+
+  private final OrganizationService            organizationService;
+
+  public GroupSynchronizationSocialProfileListener(ProfilePropertySettingsService profilePropertySettingsService,
+                                                   OrganizationService organizationService) {
+    this.profilePropertySettingsService = profilePropertySettingsService;
+    this.organizationService = organizationService;
+  }
+
+  @Override
+  public void avatarUpdated(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+  }
+
+  @Override
+  public void bannerUpdated(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+  }
+
+  @Override
+  public void contactSectionUpdated(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+
+  }
+
+  @Override
+  public void experienceSectionUpdated(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+
+  }
+
+  @Override
+  public void createProfile(ProfileLifeCycleEvent event) {
+    profileUpdated(event);
+
+  }
+
+  private void profileUpdated(ProfileLifeCycleEvent profileLifeCycleEvent) {
+    Profile profile = profileLifeCycleEvent.getProfile();
+    User user;
+    try {
+      user = organizationService.getUserHandler().findUserByName(profile.getIdentity().getRemoteId());
```

Comment:
no need to declare the user variable outside the try block

```suggestion
    try {
      User user = organizationService.getUserHandler().findUserByName(profile.getIdentity().getRemoteId());
```

Reply:
Done